File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int pick = nums.length / 2;
        
        Map<Integer, Integer> monsterMap = new HashMap<>();
        
        for (int num: nums) {
            monsterMap.put(num, monsterMap.getOrDefault(num, 0) + 1);
        }
        
        return Math.min(pick, monsterMap.keySet().size());
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    pocketmon_dict = {}
    pick = len(nums) / 2
    
    for num in nums:
        if num not in pocketmon_dict.keys():
            pocketmon_dict[num] = 1
        else:
            pocketmon_dict[num] += 1
    
    return min(pick, len(pocketmon_dict))